jpeginfo.c

```diff
@@ -384,16 +367,26 @@ void parse_jpeg_info(struct jpeg_decompress_struct *cinfo, struct jpeg_info *inf
 	info->color_depth = (int)cinfo->num_components * 8;
 	info->progressive = (cinfo->progressive_mode ? 1 : 0);
 
+	char info_str[256];
 	strncopy(info_str, (cinfo->arith_code ? "Arithmetic" : "Huffman"), sizeof(info_str));
+	char comment_str[1024];
 	comment_str[0]=0;
+	char marker_str[256];
 	marker_str[0]=0;
 
 	/* Check for special (Exif/IPTC/ICC/XMP/etc...) markers */
-	cmarker=cinfo->marker_list;
+	jpeg_saved_marker_ptr cmarker=cinfo->marker_list;
+
+	char tmp[64];
```

Comment:
Maybe even more cleaner look would be to remove this tmp variable. And add appropriate sized tmp variables below in the contexts where its used?

Reply:
I made them local to their uses. I was able to easily shrink the `snprintf` one as gcc 13 will complain if the buffer is too small. The `cmarker->data` case wasn't immediately obvious if I could make it smaller.